### PR TITLE
feat: Ability to add custom text field

### LINKED ISSUE #688: Ability to add text fields
### Feature Description

Ability for requestor to add any text filed on any place of the document for the signer to fill out. 

There is already the "Name" input. But we want to be able to add any label, not just "Name".

### Use Case

_No response_

### Proposed Solution

_No response_

### Alternatives (optional)

_No response_

### Additional Context

_No response_

### Please check the boxes that apply to this feature request.

- [ ] I have searched the existing feature requests to make sure this is not a duplicate.
- [ ] I have provided a detailed description of the requested feature.
- [ ] I have explained the use case or scenario for this feature.
- [ ] I have included any relevant technical details or design suggestions.
- [ ] I understand that this is a suggestion and that there is no guarantee of implementation.

### PR DESCRIPTION
**Pull request by anikdhabal**
fixes #688 

Loom Video:-

For SinglePlayerMode->

https://github.com/documenso/documenso/assets/81948346/cd9568af-3473-4399-93f3-ddc9625b42c9

For Main App-> 

https://github.com/documenso/documenso/assets/81948346/6f492227-9d60-486b-a273-a24f6a9b6545

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "typescript.tsdk": "node_modules/typescript/lib",
   "editor.codeActionsOnSave": {
-    "source.fixAll.eslint": true
+    "source.fixAll.eslint": "explicit"
```
Comment (adithyaakrishna) [inline .vscode/settings.json:4]: @anikdhabal Could you please revert this change if it is not needed?

File: packages/ui/primitives/document-flow/add-fields.types.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ export const ZAddFieldsFormSchema = z.object({
       nativeId: z.number().optional(),
       type: z.nativeEnum(FieldType),
       signerEmail: z.string().min(1),
+      customFieldValue: z.string().min(0),
```
Comment (adithyaakrishna) [inline packages/ui/primitives/document-flow/add-fields.types.ts:12]: ` customFieldValue: z.string().optional(),`

File: .vscode/settings.json (hunk 1 above)
Comment (anikdhabal) [reply .vscode/settings.json:4]: Yes, sure

**Comment by anikdhabal** [comment]
@dguyen check this once.

**Comment by Mythie** [comment]
This is still on my mind but we're also considering how we can make a nicer field API at the moment thus why progress on review and merging has slowed. I'll keep this updated as we move forward.

**Comment by anikdhabal** [comment]
> This is still on my mind but we're also considering how we can make a nicer field API at the moment thus why progress on review and merging has slowed. I'll keep this updated as we move forward.

Thank you Lucas🙏

**Comment by Mythie** [comment]
Closing this one out since we now kind of have the same feature, Catalin's work on advanced fields also covers the labelling and placeholder aspect. 

We still appreciate the work you did and apologise for not moving fast enough on this!
